### DISCUSSION TITLE
How to specify additional CSS

### DISCUSSION THREAD

**Discussion by larsen** [nextjournal/clerk#610] [Q&A] [answered]
I would like to use and visualize GPX data in a notebook, so I am experimenting with a custom viewer that uses [Leaflet](https://leafletjs.com/).

Adapting an [example in The Book of Clerk](https://book.clerk.vision/#loading-libraries) I was able to use `with-d3-require` to load the JS code I needed, but according to [Leaflet's quickstart guide](https://leafletjs.com/examples/quick-start/) I also have to include a CSS document. 

I could not find a way to do that from a notebook. 

I am wondering if I am pursuing a valid idea in general (trying to include extra CSS), and what would be the appropriate mechanism to allow users to do that.

Looking into Clerk's code I found a function `include-css+js` that seems pertinent (and in fact it calls `hiccup/include-css` to include the CSS for KaTeX, which is in this respect similar to Leaflet: I mean that using KaTeX implies loading some JS code _and_ a CSS). 
Perhaps I could devise a way to make `include-css+js` receive extra instructions from the user, but again I am not sure this is a sound idea.

Thank you for any guidance or hints you can provide

**Comment by zampino** [comment, marked as answer]
Hi @larsen, what we're doing in some projects of ours, is to alter the var root of the functions you mention, like: 

```clojure
(alter-var-root #'nextjournal.clerk.view/include-css+js
                  (fn [include-fn]
                    (fn [state]
                      (concat (include-fn state)
                              (list (hiccup.page/include-css "https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"))))))
```

this works well enough until we find a dedicated mechanism to allow for 3rd party css additions.

**Reply by larsen** [reply]
Thank you!

**Comment by zampino** [comment]
See also discussion in https://github.com/nextjournal/clerk/issues/67 for more context.

**Comment by mk** [comment]
https://github.com/nextjournal/clerk/issues/323 is a related issue with more info, feel free to upvote it.

**Reply by larsen** [reply]
Thank you for the extra info!
